tinygrad/tensor.py

```diff
@@ -251,10 +251,7 @@ def pad(self, arg: Tuple[Tuple[int, int], ...], value:float=0) -> Tensor:
   # ***** movement hlops *****
 
   # NOTE: using slice is discouraged and things should migrate to pad and shrink
-  def slice(self, arg:Sequence[Optional[Tuple[int, int]]]) -> Tensor:
-    arg_ = tuple([a if a is not None else (0,s) for s,a in zip(self.shape, arg)])
-    padding = tuple([(max(0, -p[0]), max(0, p[1]-self.shape[i])) for i,p in enumerate(arg_)])
-    return self.pad(padding).shrink(tuple([(p[0] + padding[i][0], p[1] + padding[i][0]) for i,p in enumerate(arg_)]))
+  def slice(self, arg: Sequence[Optional[Tuple[int, int]]]) -> Tensor: return self.pad2d(flatten(((a[1] - s, -a[0]) if a is not None else (0, 0) for s, a in zip(self.shape, arg)))[::-1])
```

Comment:
slice should not call pad2d. Can this be rewritten so pad2d calls slice?

Reply:
Current implementation is like that so I reverted my commit and added const support to both slice and pad2d